blocks/library/cover-image/index.js

```diff
@@ -0,0 +1,169 @@
+/**
+ * WordPress dependencies
+ */
+import Placeholder from 'components/placeholder';
+
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+import { registerBlockType, query } from '../../api';
+import Editable from '../../editable';
+import MediaUploadButton from '../../media-upload-button';
+
+const { text } = query;
+
+/**
+ * Returns an attribute setter with behavior that if the target value is
+ * already the assigned attribute value, it will be set to undefined.
+ *
+ * @param  {string}   align Alignment value
+ * @return {Function}       Attribute setter
+ */
+function toggleAlignment( align ) {
+	return ( attributes, setAttributes ) => {
+		const nextAlign = attributes.align === align ? undefined : align;
+		setAttributes( { align: nextAlign } );
+	};
+}
+
+const editMediaLibrary = ( attributes, setAttributes ) => {
+	const frameConfig = {
+		frame: 'post',
+		title: wp.i18n.__( 'Change cover image' ),
+		button: {
+			text: wp.i18n.__( 'Select' ),
+		},
+	};
+
+	const editFrame = wp.media( frameConfig );
+	function updateFn( model ) {
+		setAttributes( {
+			url: model.single().attributes.url,
+			image: model.single().attributes,
+		} );
+	}
+
+	editFrame.on( 'insert', updateFn );
+	editFrame.open( 'cover-image' );
+};
```

Comment:
This is the second block allowing to edit a media. I wonder if should try to factorize this in `MediaUploadButton` or similar. This could be done in a separate PR.